File: python/smallestIndex.py
# Time -> 0(Nd) Space -> 0(d)
def smallestIndex(nums: list[int]) -> int:
    for idx, val in enumerate(nums):
        currSum = 0
        for v in str(val):
            currSum += int(v)
        if idx == currSum:
            return idx

    return -1


print(smallestIndex([1, 3, 2]))
print(smallestIndex([1, 10, 11]))
print(smallestIndex([1, 2, 3]))

File: go/SmallestIndex/SmallestIndex.go
package main

import "fmt"

// Time -> 1(Nd) Space -> 0(d)
func smallestIndex(nums []int) int {
	for idx := range nums {
		val := fmt.Sprint(nums[idx])
		sum := 0
		for i := range val {
			sum += int(val[i] - '0')
		}
		if sum == idx {
			return idx
		}
	}

	return -1
}

func main() {
	fmt.Println(smallestIndex([]int{1, 3, 2}))
	fmt.Println(smallestIndex([]int{1, 10, 11}))
	fmt.Println(smallestIndex([]int{1, 2, 3}))
}
